pkg/debuginfo/debuginfo.go

```diff
@@ -243,18 +243,30 @@ func (di *Extractor) extract(ctx context.Context, buildID string, file string) (
 		level.Debug(di.logger).Log("msg", "failed to determine if binary is a Go binary", "path", file, "err", err)
 	}
 
+	sections, err := sectionsToRemove(file)
+	if err != nil {
+		level.Debug(di.logger).Log("msg", "failed to determine sections to remove", "path", file, "err", err)
+	}
+
 	var (
 		cmd           *exec.Cmd
 		debugInfoFile string
+		cleanup       func() error
 	)
 	switch {
 	case hasDWARF:
-		cmd, debugInfoFile = di.useStrip(ctx, tmpDir, file)
+		cmd, debugInfoFile, cleanup = di.useStrip(ctx, tmpDir, file, sections)
 	case isGo:
-		cmd, debugInfoFile = di.useObjcopy(ctx, tmpDir, file)
+		cmd, debugInfoFile, cleanup = di.useObjcopy(ctx, tmpDir, file, sections)
 	default:
-		cmd, debugInfoFile = di.useStrip(ctx, tmpDir, file)
+		cmd, debugInfoFile, cleanup = di.useStrip(ctx, tmpDir, file, sections)
 	}
+	defer func() {
```

Comment:
Why defer the cleanup?

Reply:
Also, why return a function at all? It seems that `useStrip` is the only method returning a cleanup function that does anything. Why not leave that defer in `useStrip` which cleans the interim file itself?

Reply:
Ack, ok I see now, that makes sense.

However, is there any reason to not simply execute the command in the function we call and return the output directly? It seems like the caller (`di.extract` in this case) is only concerned with the output and doesn't need a handle to the `*exec.Command` pointer.

Reply:
I think this we can do :+1: Thanks for the suggestion.